### ISSUE TITLE
Parametrized KPP

### ISSUE THREAD

**Issue by navidcy** [dhruvbhagtani/sfc-perturbation-expts#14]
I was thinking about the fact that was discussed in the zoom today that the same parameters for the parametrised KPP seem to work equally as well for the 1 deg model and the 0.25 deg.

Don't these two models differ in the mixed-layer depth? The parametrisation only "looks" at the friction velocity, i.e., the surface wind stress. Thus, ignoring some minor differences in wind stress due to ocean flow, it should be the same for the two models, right? So how come it works as well for both models? Am I missing something?

@dhruvbhagtani, @AndyHoggANU, @rmholmes

**Comment by dhruvbhagtani** [comment]
Good point Navid. The two models do differ a bit with regards to the KPP depth. As Ryan pointed out today, it'll be better to have a comparison of our present 1 degree models with the quarter degree outputs. I'll upload it here as soon as I get that notebook ready.

**Comment by dhruvbhagtani** [comment]
<img width="1217" alt="Screen Shot 2021-10-12 at 1 41 19 pm" src="https://user-images.githubusercontent.com/29700296/136917828-13ee9ae0-5698-46db-a093-9ea6fa2b2233.png">
As promised @navidcy, here are the four experiments: (i) Control_1deg, (ii) Param_1deg, (iii) Control_025deg, and (iv) Param_025deg. Our parameterisation does perform differently for the 1 degree and 025 degree experiments, but the differences are overshadowed by the error in our parameterisation, which is around 5-15 %, depending on the basin we are observing.

**Comment by dhruvbhagtani** [comment]
I only have the first 15 years in the graph above because I ran the `1 degree control` for 15 years only, and when I realised that the `1 degree` parameterisation was doing okay, I stopped running it.

**Comment by navidcy** [comment]
I'm wondering whether this disagreement will come and bite us later on... But let me not worry you (given that others are ok).

These are very broad metrics, I remember some movies which showed spatial differences. Also I don't recall whether the differences seems to get larger or smaller as the simulations evolved longer...

**Comment by navidcy** [comment]
I saw a branch `Parametrized-KPP`, is this related? Why don't we open a PR and discuss is and when we all are happy merge it to `main`?

**Comment by AndyHoggANU** [comment]
Yep, I think I am OK with this. There are differences, but the key thing is that we are going to hit these expts with huge forcing changes which will swamp the param differences. And then, we will compare like with like. 

One thing, if it’s not too much trouble I would consider running the 1deg control for longer so we can document the differences of the param in case we need it later.

**Comment by dhruvbhagtani** [comment]
@navidcy the parameterising-KPP-shear branch is related. It's in a branch in progress, as I have to compare them with the 0.25 degree outputs too. When that is done, we can merge.

Sure @AndyHoggANU, I'll run it till the full 100 years?

**Comment by rmholmes** [comment]
I agree with all of the above.

My only doubt is exactly what the figures above represent. I think I remember discussing this before - but it'd be great to actually plot the x,y structure (perhaps even as a bias w.r.t. the control) so that we know what we are seeing is not the result of a cancellation of large changes in different regions (esp. given that the BL depth distribution is far from normal and so the mean may not be a great representative statistic).

**Comment by navidcy** [comment]
Could we also write here the parametrization for the parametrization we used? I mean `u_resolved = f(τ/ρ)`.

**Comment by dhruvbhagtani** [comment]
Oops, I didn't send this movie before. It is a daily fractional error (hblt_cont - hblt_pram)/hblt_cont for the 15th year after the simulation has started. The errors are less. I can also create one for the last (100th year) after my control has also run for 100 years.
[hblt_diff.m4v.zip](https://github.com/dhruvbhagtani/varying-surface-forcing/files/7328959/hblt_diff.m4v.zip)
.

**Comment by dhruvbhagtani** [comment]
To be honest, I myself didn't have a good look at them till now, but isn't there something weird happening in the equator?

**Comment by dhruvbhagtani** [comment]
Yes, the parameterisation is given by:
`dVsq = f(ustar, z)`, 
where `dVsq` is the resolved velocity shear and `ustar = (τ/ρ)^(1/2)` is the frictional velocity at a given latitude and longitude. Should I also write down the exact equation here @navidcy? It is quite lengthy.

**Comment by navidcy** [comment]
I just thought perhaps we could think about it... Perhaps we shouldn't, we should just move on.

**Comment by dhruvbhagtani** [comment]
Actually, the differences we observe are related to the inter annual variability. Have a look at the attached plot. This is for the 0.25 degree run. The major differences are in the winter season, where our BL depth overshoots
<img width="1158" alt="Screen Shot 2021-10-12 at 4 33 36 pm" src="https://user-images.githubusercontent.com/29700296/136944374-5facfa54-9243-4e3a-8a8b-355d68005f11.png">
.

**Comment by dhruvbhagtani** [comment]
Here is the comparison of some diagnostics between the 1deg and 025deg model: https://github.com/dhruvbhagtani/varying-surface-forcing/blob/main/SH_winds/Diagnostics.ipynb

**Comment by dhruvbhagtani** [comment]
The surface salinity values for `0.25 degree` winds in SH case stand out so much! The entire northern hemisphere is so much fresher than other three runs.

<img width="1687" alt="Screen Shot 2021-10-14 at 3 34 54 pm" src="https://user-images.githubusercontent.com/29700296/137296932-bb994b39-6fe2-42e0-856f-03a615c279cd.png">

**Comment by dhruvbhagtani** [comment]
There is a much stronger heat flux in the sub-polar and western boundary regions in North Atlantic in the 0.25 degree run than in the 1 degree run. The western boundary is also much stronger, so is the subpolar gyre. Although the increase in circulation could be due to a variety of factors, it is worth mentioning.
<img width="1771" alt="Screen Shot 2021-10-14 at 3 36 47 pm" src="https://user-images.githubusercontent.com/29700296/137297789-482aae6a-867a-49fe-b1f8-9e3a4833fc87.png">

<img width="1788" alt="Screen Shot 2021-10-14 at 3 36 33 pm" src="https://user-images.githubusercontent.com/29700296/137297768-da857dca-3164-462f-a54a-c753ef5521b7.png">

**Comment by dhruvbhagtani** [comment]
The biggest differences in heat flux components are in the sensible and latent heat fluxes. In the CICE, they use an empirical relation given by:
1. Latent heat = 𝜌 𝐶𝑒 𝑢10 (𝑞𝑠 − 𝑞𝑎) 𝐿, where 𝐶𝑒 is the transfer coefficient for latent heat, u10 is the wind velocity 10 above the sea surface, 𝑞𝑠 is 98% of the saturated specific humidity at the SST, 𝑞𝑎 is the measured specific humidity, and 𝐿 is the latent heat of vaporisation/evaporation.
2. Sensible heat = 𝜌 𝑐𝑝 𝐶ℎ 𝑢10 (𝑇𝑠 − 𝑇𝑎), where 𝑐𝑝 is the specific heat capacity of seawater, 𝐶ℎ is the transfer coefficient for sensible heat, 𝑇𝑠 is the SST and 𝑇𝑎 is the air temperature above the sea surface.

Here are the plots for all four heats: sensible, latent, shortwave and long wave:

<img width="1623" alt="Screen Shot 2021-10-14 at 3 57 55 pm" src="https://user-images.githubusercontent.com/29700296/137300532-b6022848-c0c4-490a-aaf2-c53f4e59bc06.png">

**Comment by dhruvbhagtani** [comment]
Finally, for winds in Southern Hemisphere only, the overturning in the 1 degree case extends far north and closer to the surface than the 0.25 degree case.

<img width="938" alt="Screen Shot 2021-10-14 at 4 00 49 pm" src="https://user-images.githubusercontent.com/29700296/137301117-f65fd8e5-de53-480d-82fe-6ccb8cb96e50.png">

**Comment by AndyHoggANU** [comment]
Curious that the AMOC seems to last longer in this case -- can we see maps of the KPP depth differences and the heat flux components? Averaged over the last decade should be fine ...

**Comment by dhruvbhagtani** [comment]
Sure, here they are:

It's interesting to see that the `hblt` differences are more amplified in the `025deg` run. Similarly, there is a weak heat loss in the winds in SH `1deg` case in the sub polar North Atlantic, which is missing in the `025deg` run.

<img width="1153" alt="Screen Shot 2021-10-16 at 2 00 04 pm" src="https://user-images.githubusercontent.com/29700296/137587324-f4dc40d6-5df9-4cef-af05-6d244fad32a4.png">
<img width="1000" alt="Screen Shot 2021-10-16 at 5 54 31 pm" src="https://user-images.githubusercontent.com/29700296/137587332-0ff90a6a-b419-4127-8123-f2e6f84327a2.png">

**Comment by dhruvbhagtani** [comment]
Here is another comparison, this time between the control and no-stres s

**Comment by dhruvbhagtani** [comment]
I made another comparison, this time between the no-stress and control case, for 1 degree and 0.25 degree simulations. Here it is, if interested:
https://github.com/dhruvbhagtani/varying-surface-forcing/tree/main/Nostress
